src/async.js

```diff
@@ -162,15 +177,24 @@ Async.prototype._drainQueue = function(queue) {
         var receiver = queue.shift();
         var arg = queue.shift();
         fn.call(receiver, arg);
+        maxItemsToProcess--;
     }
+
+    return maxItemsToProcess;
 };
 
-Async.prototype._drainQueues = function () {
+Async.prototype._drainQueues = function (maxItemsToProcess) {
     ASSERT(this._isTickUsed);
-    this._drainQueue(this._normalQueue);
+    maxItemsToProcess = this._drainQueue(this._normalQueue, maxItemsToProcess);
     this._reset();
     this._haveDrainedQueues = true;
-    this._drainQueue(this._lateQueue);
+    this._drainQueue(this._lateQueue, maxItemsToProcess);
```

Comment:
I think in my original PR, batchSize is only used for the normal Queue and not the late one. I forget why. I'm trying to remember what the late queue contains - I think it's where fulfillHandlers/rejectHandlers are processed. It may be safe to break that up into batches, but I'm not sure. Maybe hold off on using batchSize for the _lateQueue for now, and file a task to look into using it.